### PR TITLE
docs: update guides + clean-up basic crawler api docs

### PR DESCRIPTION
**Pull request by AndreyBykov**
- Went throught `got-scpraing` guide and fixed some typos, `you -> we`, etc. Also move it after `avoid blocking` in the side menu.
- Went through Basic Crawler API docs, fixed/updated some descriptions, added some links, etc.

Please check it out if you would have the internet (maybe even render the docs) - if it's all good - will do the rest tomorrow (or better to say today :/)

### CODE REVIEW

File: docs/guides/got_scraping.mdx [hunk 1]
```diff
@@ -1,13 +1,15 @@
 ---
 id: got-scraping
-title: Got Scraping guide
+title: Got Scraping
 ---
 
 ## Intro
 
-If you're using Basic Crawler, you have to send requests manually. In order to do this, you need to call `sendRequest()` like this:
+If we're using Basic Crawler, we have to send requests manually. In order to do this, We need to call `sendRequest()` like this:
```
Comment (B4nan) [inline docs/guides/got_scraping.mdx:8]:
```suggestion
When using `BasicCrawler`, we have to send the requests manually. In order to do this, we can use the context-aware `sendRequest()` function:
```

File: docs/guides/got_scraping.mdx [hunk 2]
```diff
@@ -49,35 +51,37 @@ async sendRequest(overrideOptions?: GotOptionsInit) => {
 
 By default, it's the URL of current task. However you can override this with a `string` or a `URL` instance if necessary.
 
-See the [Got documentation](https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#url) for more details.
+*More details: [Got documentation](https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#url).*
```
Comment (B4nan) [inline docs/guides/got_scraping.mdx:54]: maybe its better this way? idk

```suggestion
*More details in [Got documentation](https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#url).*
```

File: docs/guides/got_scraping.mdx [hunk 3]
```diff
@@ -135,26 +142,29 @@ const crawler = new BasicCrawler({
...
+- *[Got documentation](https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#cookiejar)*
+- *[Tough Cookie documentation](https://github.com/salesforce/tough-cookie#cookiejarstore-options)*
 
 ### `retry.limit`
 
-This option specifes the maximum number of Got retries.
+This option specifies the maximum number of `Got` retries.
 
-By default, `retry.limit` is set to `0`. This is because `crawlee` has its own (complicated enough) retry management.
+By default, `retry.limit` is set to `0`. This is because `Crawlee` has its own (complicated enough) retry management.
```
Comment (B4nan) [inline docs/guides/got_scraping.mdx:153]:
```suggestion
By default, `retry.limit` is set to `0`. This is because Crawlee has its own (complicated enough) retry management.
```

File: docs/guides/request_storage.mdx [hunk 4]
```diff
@@ -145,4 +145,4 @@ import { purgeDefaultStorages } from 'crawlee';
 await purgeDefaultStorages();
 ```
 
-Calling this function will clean up the default request storage directory (and also the request list stored in default key-value store). This is a shortcut for running (optional) `purge` method on the <ApiLink to="types/interface/StorageClient">`StorageClient`</ApiLink> interface, in other words it will call the `purge` method of the underlying storage implementation we are currently using. In addition, this method will make sure the storage is purged only once for a given execution context, so it is safe to call it multiple times.
+Calling this function will clean up the default request storage directory (and also the request list stored in default key-value store). This is a shortcut for running (optional) `purge` method on the <ApiLink to="core/interface/StorageClient">`StorageClient`</ApiLink> interface, in other words it will call the `purge` method of the underlying storage implementation we are currently using. In addition, this method will make sure the storage is purged only once for a given execution context, so it is safe to call it multiple times.
```
Comment (B4nan) [inline docs/guides/request_storage.mdx:148]: just curious, the link was not working before? because it was valid, i just added reexport yesterday to the core package. i guess it again (wrongly) marked the one in types packages as reexport? we should probably report this to typedoc, i'd say there will be some issue about this already

**Review by B4nan — APPROVED** [review]
looking good, i'll merge it right ahead and we can again improve later via another PR

File: docs/guides/got_scraping.mdx (hunk 2 above)
Comment (AndreyBykov) [reply docs/guides/got_scraping.mdx:54]: I was thinking about this one, both sounds good to me, will update to `in` with next PR :)

File: docs/guides/request_storage.mdx (hunk 4 above)
Comment (AndreyBykov) [reply docs/guides/request_storage.mdx:148]: Yeah, with `types/interface/StorageClient` it returned `Page does not exist` or something like that :/

And yeah - in `types` it has it only in `References` as `Re-exports [StorageClient](https://crawlee.dev/api/core/interface/StorageClient)`

File: docs/guides/got_scraping.mdx (hunk 2 above)
Comment (szmarczak) [inline docs/guides/got_scraping.mdx:54]: What's the rationale behind this change? I don't mind, just asking.

Comment (AndreyBykov) [reply docs/guides/got_scraping.mdx:54]: Originally @B4nan mentioned this `you` -> `we` thing on Slack, so basically I tried to change `[you] see more details` to something less direct/addressed 🤷‍♂️

But in the end, we're not doing it...

Comment (szmarczak) [reply docs/guides/got_scraping.mdx:54]: 😂 Ok I get it now :D That's kinda dumb because there's often `Read more` in other projects' docs.

Like I suggested, `you` refers explicitly to the developer reading the docs, so if *they* are given an option then the steps depend on *their* path *they* have chosen. So `you` would be appropriate. https://docs.oracle.com/en/java/

> Java SE lets you develop

There's an implied choice. You can choose Java or any other language.

OTOH `we` refers to both the developer and the project, there may be a thing that applies to both persons. For example a tutorial. https://docs.oracle.com/search/?q=we&category=java&product=en%2Fjava

> We will run the HelloWorld sample

Comment (AndreyBykov) [reply docs/guides/got_scraping.mdx:54]: I think that's exactly what we're doing in the end 😄 But this PR was already merged before :/
